### DISCUSSION TITLE
New Programmer, Old D2 player

### DISCUSSION THREAD

**Discussion by MatthewDanh** [nxtlo/aiobungie#382] [General]
I am a new programmer, by new I mean I have been studying various languages like python, java, C++, Web Dev (HTML, CSS, JS), and IT for school. I have the skills now to import and write enough python where I feel confident tackling a project without a tutorial. I also have an API key from bungie, but I do not know what kind of project to do that still is in my scope. Being "Intermediate" is tough because it is either too easy, or I do not know enough to get out of trouble when I run into it. So, any advice would be greatly appreciated.

**Comment by nxtlo** [comment]
hi @MatthewDanh, sorry for the late reply, i'm really not sure if this question is related to aiobungie or just in general? so i will just assume that you want to start a project using aiobungie and you want ideas.

since you know web dev, maybe start with a simple web application, the stack should be simple, just [FastAPI](https://github.com/tiangolo/fastapi) since aiobungie is `asynchronous` and aiobungie it self as the API adapter.

but in general, it really is just programming, you learn by building stuff, if you learn Python, then you'll also learn aiobungie for free. so the core foundation here is learning Python and its idiomatics.

so if you want to use this library you need to know a couple of things:

* [`async` `await` syntax](https://docs.python.org/3/library/asyncio-task.html#coroutine), it is crucial to understand this since aiobungie is built on-top of `asyncio`
* [concurrency](https://medium.com/@georgexwee/concurrency-in-python-asyncio-eb342a429e23), this is also important when you write *async* code
* [design patterns](https://www.geeksforgeeks.org/python-design-patterns/), not as important, but its always a good thing to write good interfaces

thats really it, if you learn those couple things, you also learn aiobungie with it for free,
you just need to know what part of Bungie's API you want to use.
